src/jvm/main/org/jetbrains/kotlinx/lincheck/execution/ExecutionResult.kt

```diff
@@ -13,7 +13,9 @@ import org.jetbrains.kotlinx.lincheck.*
 
 /**
  * This class represents a result of [ExecutionScenario] execution.
- * All the result parts should have the same dimensions as the scenario.
+ * If the execution is completed, all the result parts should have the exact dimensions of the scenario.
```

Comment:
Let us allow `ExecutionResult` to store `Result?` and document that `null` denotes an operation that haven't finished.

Reply:
Now, I decided to store `NotFinished` and `NotStarted` results